Homework 1 on SmartHome

DaniilVyatkin opened the pull request:
Sorry for being so late - I am only starting out with Java:(

DaniilVyatkin committed:
Homework 1 on SmartHome

a-ivanov commented on src/main/java/ru/sbt/mipt/oop/SmartHomeInitializer.java:
```diff
@@ -0,0 +1,17 @@
+
+public class SmartHomeInitializer {
+
+        public static SmartHome initializeSmartHome() throws IOException{
```
https://github.com/iagoshkov/smart-home-2020/blob/master/STUDENT_GUIDE.md#%D0%BD%D0%B5-%D0%B8%D1%81%D0%BF%D0%BE%D0%BB%D1%8C%D0%B7%D1%83%D0%B9%D1%82%D0%B5-%D1%81%D1%82%D0%B0%D1%82%D0%B8%D1%87%D0%B5%D1%81%D0%BA%D0%B8%D0%B5-%D0%BC%D0%B5%D1%82%D0%BE%D0%B4%D1%8B-%D0%B2-%D0%B1%D0%B8%D0%B7%D0%BD%D0%B5%D1%81-%D0%BB%D0%BE%D0%B3%D0%B8%D0%BA%D0%B5

a-ivanov replied:
Добавьте интерфейс для чтения состояния дома.

a-ivanov commented on src/main/java/ru/sbt/mipt/oop/EventHandlingLoop.java:
```diff
@@ -0,0 +1,86 @@
+
+public class EventHandlingLoop {
+
+    public static void handleEvents(SmartHome smartHome) {
```
Класс не сильно отличается от исходного кода. Разделите его по ответственностям на разные обработчики, сейчас нарушение SRP.

a-ivanov commented on src/main/java/ru/sbt/mipt/oop/EventHandlingLoop.java:
```diff
@@ -0,0 +1,86 @@
+
+    public static void handleEvents(SmartHome smartHome) {
+        // начинаем цикл обработки событий
+        while (true) {
```
https://github.com/iagoshkov/smart-home-2020/blob/master/STUDENT_GUIDE.md#%D0%BF%D0%B5%D1%80%D0%B5%D0%BD%D0%B5%D1%81%D0%B8%D1%82%D0%B5-%D0%BB%D0%BE%D0%B3%D0%B8%D0%BA%D1%83-%D1%86%D0%B8%D0%BA%D0%BB%D0%B0-%D0%BE%D0%B1%D1%80%D0%B0%D0%B1%D0%BE%D1%82%D0%BA%D0%B8-%D1%81%D0%BE%D0%B1%D1%8B%D1%82%D0%B8%D0%B9-%D0%B2-%D0%BE%D1%82%D0%B4%D0%B5%D0%BB%D1%8C%D0%BD%D1%8B%D0%B9-%D0%BA%D0%BB%D0%B0%D1%81%D1%81

a-ivanov commented on src/main/java/ru/sbt/mipt/oop/EventHandlingLoop.java:
```diff
@@ -0,0 +1,86 @@
+                    } else {
+                        System.out.println(stringToPrint + "closed.");
+                    }
+                    additionalActionInCertainRoomsOnDoorEvent(event, room, smartHome);
```
https://github.com/iagoshkov/smart-home-2020/blob/master/STUDENT_GUIDE.md#%D1%80%D0%B0%D0%B7%D0%B4%D0%B5%D0%BB%D1%8F%D0%B9%D1%82%D0%B5-%D0%B8%D0%B7%D0%BC%D0%B5%D0%BD%D0%B5%D0%BD%D0%B8%D0%B5-%D0%BE%D0%B1%D1%8A%D0%B5%D0%BA%D1%82%D0%BE%D0%B2-%D0%B8-%D1%81%D1%86%D0%B5%D0%BD%D0%B0%D1%80%D0%B8%D0%B8-%D1%83%D0%BC%D0%BD%D0%BE%D0%B3%D0%BE-%D0%B4%D0%BE%D0%BC%D0%B0

DaniilVyatkin committed:
Homework 1 corrections.
Поправил места, к которым были замечания:
1. Добавил и реализовал интерфейс для чтения (и еще интерфейс для записи) состояния дома
2. Разделил класс EventHandlingLoop на отдельные обработчики событий (для света, двери и входной двери)
3. Добавил и реализовал интерфейс для генерации событий
4. Добавил интерфейс обработчиков событий и реализовал его для обработчиков событий света, двери и входной двери

a-ivanov commented on src/main/java/ru/sbt/mipt/oop/EventHandlingLoop.java:
```diff
@@ -0,0 +1,23 @@
+
+public class EventHandlingLoop {
+
+    public static void handleEvents(SmartHome smartHome, EventGenerator eventGenerator) {
```
Не делайте статических методов, здесь это не нужно.

a-ivanov commented on src/main/java/ru/sbt/mipt/oop/EventHandlingLoop.java:
```diff
@@ -0,0 +1,23 @@
+
+    public static void handleEvents(SmartHome smartHome, EventGenerator eventGenerator) {
+
+        ArrayList<EventHandler> eventHandlers = new ArrayList<>(
```
Создайте инстанс класса и передавайте ему список обработчиков через конструктор.
Сделайте поле `List<EventHandler> eventHandlers;`, передавать в него можно `Arrays.asList(...)`.

DaniilVyatkin committed:
Homework 1 more corrections.
Поправил места, к которым были замечания:
1. Убрал статический метод из EventHandlingLoop
2. Добавил в Application.main создание экземпляра EventHandlingLoop с передачей ему листа EventHadler'ов
